packages/next/src/server/lib/router-utils/route-types-utils.ts

```diff
@@ -120,6 +133,36 @@ function isCanonicalRoute(route: string) {
   return true
 }
 
+/**
+ * Resolves an intercepting route to its canonical equivalent
+ * Example: /gallery/test/(..)photo/[id] -> /gallery/photo/[id]
+ */
+function resolveInterceptingRoute(route: string): string {
+  const segments = route.split('/').filter(Boolean)
+  const resolved: string[] = []
+
+  for (let i = 0; i < segments.length; i++) {
```

Comment:
Could we re-use the existing normalizing here instead https://github.com/vercel/next.js/blob/feat/type-validation-new/packages/next/src/shared/lib/router/utils/interception-routes.ts#L22

Reply:
Done!